File: app/src/main/java/com/jgw/delingha/module/login/model/LoginModel.java
package com.jgw.delingha.module.login.model;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.jgw.common_library.http.Resource;
import com.jgw.common_library.http.rxjava.CustomObserver;
import com.jgw.common_library.livedata.ValueKeeperLiveData;
import com.jgw.common_library.utils.LogUtils;
import com.jgw.common_library.utils.StringUtil;
import com.jgw.delingha.bean.LoginBean;
import com.jgw.delingha.bean.UserBean;
import com.jgw.delingha.network.HttpUtils;
import com.jgw.delingha.network.api.ApiService;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.disposables.Disposable;

public class LoginModel {

    public LiveData<Resource<UserBean>> login(LoginBean bean) {
        LogUtils.xswShowLog("phone = " + bean.getMobile() + " password = " + StringUtil.md5Decode(bean.getPwd()));
        MutableLiveData<Resource<UserBean>> liveData = new ValueKeeperLiveData<>();
        Map<String, Object> map = new HashMap<>();
        map.put("account", bean.getMobile());
        map.put("password", StringUtil.md5Decode(bean.getPwd()));
        map.put("tokenTimeout", 3600 * 24 * 30);
        HttpUtils.getGatewayApi(ApiService.class)
                .login(map)
                .compose(HttpUtils.applyMainSchedulers())
                .subscribe(new CustomObserver<UserBean>() {
                    @Override
                    public void onNext(UserBean userBean) {
                        liveData.postValue(new Resource<>(Resource.SUCCESS, userBean, ""));
                    }

                    @Override
                    public void onSubscribe(Disposable d) {
                        liveData.postValue(new Resource<>(Resource.LOADING, null, ""));
                    }

                    @Override
                    public void onError(Throwable e) {
                        super.onError(e);
                        liveData.postValue(new Resource<>(Resource.ERROR, null, ""));
                    }
                });
        return liveData;
    }

}

File: app/src/main/java/com/ggb/nirvanahappyclub/module/login/LoginModel.kt
package com.ggb.nirvanahappyclub.module.login

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.ggb.common_library.base.adapter.CustomRecyclerAdapter
import com.ggb.common_library.http.Resource
import com.ggb.common_library.http.rxjava.CustomObserver
import com.ggb.common_library.livedata.ValueKeeperLiveData
import com.ggb.nirvanahappyclub.bean.ArticleContentBean
import com.ggb.nirvanahappyclub.bean.IndexArticleInfoBean
import com.ggb.nirvanahappyclub.bean.SimpleUserInfo
import com.ggb.nirvanahappyclub.network.HttpUtils
import com.ggb.nirvanahappyclub.network.api.ApiService
import io.reactivex.disposables.Disposable

class LoginModel {

    private val userLoginByPwdLiveData: MutableLiveData<Resource<String>> = ValueKeeperLiveData()

    fun getUserLoginByPwdLiveData(map:HashMap<String,Any>): LiveData<Resource<String>> {

        HttpUtils.getGatewayApi(ApiService::class.java)
            .login(map)
            .compose(HttpUtils.applyMainSchedulers())
            .subscribe(object : CustomObserver<String>() {

                override fun onSubscribe(d: Disposable) {
                    super.onSubscribe(d)
                    userLoginByPwdLiveData.value = Resource(Resource.LOADING, null, "")
                }

                override fun onNext(s: String) {
                    userLoginByPwdLiveData.value = Resource(Resource.SUCCESS, s, "")
                }

                override fun onError(e: Throwable) {
                    super.onError(e)
                    userLoginByPwdLiveData.value = Resource(Resource.ERROR, null, e.message)
                }
            })
        return userLoginByPwdLiveData
    }

    private val userInfoLiveData: MutableLiveData<Resource<SimpleUserInfo>> = ValueKeeperLiveData()

    fun getUserInfoLiveData(xx:String): LiveData<Resource<SimpleUserInfo>> {

        HttpUtils.getGatewayApi(ApiService::class.java)
            .userInfo
            .compose(HttpUtils.applyMainSchedulers())
            .subscribe(object : CustomObserver<SimpleUserInfo>() {

                override fun onSubscribe(d: Disposable) {
                    super.onSubscribe(d)
                    userInfoLiveData.value = Resource(Resource.LOADING, null, "")
                }

                override fun onNext(info: SimpleUserInfo) {
                    userInfoLiveData.value = Resource(Resource.SUCCESS, info, "")
                }

                override fun onError(e: Throwable) {
                    super.onError(e)
                    userInfoLiveData.value = Resource(Resource.ERROR, null, e.message)
                }
            })
        return userInfoLiveData
    }

}